.github/workflows/multi-arch-build.yaml

```diff
@@ -44,103 +52,149 @@ jobs:
+          echo '::set-output name=push::true'
+
+      # This is substantially the same as the above step, except the
+      # $CONTAINERS_QUAY_REGISTRY is used and the "testing"
+      # flavor is never pushed.
+      - name: Generate containers reg. image FQIN(s)
+        if: matrix.source != 'testing'
+        id: containers_reg
+        run: |
+          if [[ "${{ matrix.source }}" == 'stable' ]]; then
+            VERSION='v${{ steps.sniff_test.outputs.version }}'
+            # workaround vim syntax-highlight bug: '
+            ALLTAGS=$(skopeo list-tags \
+                      docker://$CONTAINERS_QUAY_REGISTRY/buildah | \
+                      jq -r '.Tags[]')
+
+            # New image? Push quay.io/containers/buildah:vX.X.X and :latest
+            if ! fgrep -qx "$VERSION" <<<"$ALLTAGS"; then
+              FQIN="$CONTAINERS_QUAY_REGISTRY/buildah:$VERSION,$CONTAINERS_QUAY_REGISTRY/buildah:latest"
+            else # Not a new version-tagged image, but contents may be updated
+              FQIN="$CONTAINERS_QUAY_REGISTRY/buildah:latest"
+            fi
+          elif [[ "${{ matrix.source }}" == 'upstream' ]]; then
+            FQIN="$CONTAINERS_QUAY_REGISTRY/buildah:latest"
+          else
+            echo "::error::Unknown matrix item '${{ matrix.source }}'"
+            exit 1
           fi
+          echo "::warning::Pushing $FQIN"
+          echo "::set-output name=fqin::${FQIN}"
+          echo '::set-output name=push::true'
 
-          # Have labels environment variable to use it in the further steps.
-          # Env variable is used to store multi-line value
-          echo "LABELS<<EOF" >> $GITHUB_ENV
-          echo "org.opencontainers.image.source=${{ github.event.repository.html_url }}" >> $GITHUB_ENV
-          echo "org.opencontainers.image.revision=${{ github.sha }}" >> $GITHUB_ENV
-          echo "org.opencontainers.image.created=$(date -u --iso-8601=seconds)" >> $GITHUB_ENV
-          echo "EOF" >> $GITHUB_ENV
+      - name: Define LABELS multi-line env. var. value
+        run: |
+          # This is a really hacky/strange workflow idiom, required
+          # for setting multi-line $LABELS value for consumption in
+          # a future step.
+          # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings
+          cat << EOF | tee -a $GITHUB_ENV
+          LABELS<<DELIMITER
+          org.opencontainers.image.source=https://github.com/${{ github.repository }}.git
+          org.opencontainers.image.revision=${{ github.sha }}
+          org.opencontainers.image.created=$(date -u --iso-8601=seconds)
+          DELIMITER
+          EOF
 
       # Separate steps to login and push for buildah and containers quay
       # repositories are required, because 2 sets of credentials are used and `docker
-      # login` as well as `podman login` do not support having 2 different
+      # login` as well as `buildah login` do not support having 2 different
```

Comment:
FWIW, we were talking about this in a meeting earlier today, two or more sets of creds is likely to be allowed in the not too distant future.